libparsec/crates/client/src/workspace/store/prevent_sync_pattern.rs

```diff
@@ -0,0 +1,117 @@
+use std::sync::Arc;
+
+use libparsec_platform_storage::workspace::{UpdateManifestData, WorkspaceStorage};
+use libparsec_types::{
+    prelude::{anyhow, thiserror},
+    DataError, LocalDevice, LocalFolderManifest, LocalWorkspaceManifest, SecretKey,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ApplyPreventSyncPatternError {
+    #[error("Decryption error")]
+    DecryptionError,
+    #[error(transparent)]
+    Internal(anyhow::Error),
+}
+
+pub(super) async fn ensure_prevent_sync_pattern_applied_to_wksp(
+    storage: &mut WorkspaceStorage,
+    device: Arc<LocalDevice>,
+) -> Result<(), ApplyPreventSyncPatternError> {
+    const PAGE_SIZE: u32 = 1000;
+
+    let (pattern, fully_applied) = storage
+        .get_prevent_sync_pattern()
+        .await
+        .map_err(ApplyPreventSyncPatternError::Internal)?;
```

Comment:
Should we retrieve the pattern from the DB, or add it as a parameters of the function ?

Reply:
The pattern should be passed to this function (this way we have a single way to set the pattern and ensure it is applied)

Reply:
done